api/buckets/service.go

```diff
@@ -2131,18 +2214,46 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 				return nil, fmt.Errorf("there is an in progress archive")
 			// Case 1.c.
 			case ffs.Failed, ffs.Canceled:
-				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithOverride(true))
+				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithStorageConfig(storageConfig), powc.WithOverride(true))
 				if err != nil {
 					return nil, fmt.Errorf("pushing config: %s", err)
 				}
 			default:
 				return nil, fmt.Errorf("unexpected current archive status: %d", ba.Archives.Current.JobStatus)
 			}
 		} else { // Case 2.
-			jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+			res, err := s.PGClient.FFS.GetStorageConfig(ctxFFS, oldCid)
```

Comment:
Logic here, in the case of a second (or more) archive, is to get the current `StorageConfig` and compare it to the new `StorageConfig`. If they are the same, use `Replace`. If not, use `Remove` and `PushConfig`.